### DISCUSSION TITLE
Pairing swup with other html-first libraries

### DISCUSSION THREAD

**Discussion by aolko** [swup/swup#883] [General]
How well swup pairs with html-first libraries that have some form of transitions, like htmlx or unpoly?

**Comment by hirasso** [comment]
Hi @aolko , thank you for your question!

While it's certainly possible to build something where swup would work nicely together with htmx or unpoly, I wouldn't recommend it. There are some functionalities in both these libraries that would need very careful implementation to prevent collisions with swup, as they overlap functionality-wise. [`hx-boost`](https://htmx.org/attributes/hx-boost/), for example, is a basic implementation of what swup does (history mutation, conversion of a classic website to a Single Page App) and would certainly collide with it. 

Before I landed at using swup in all my projects, I investigated htmx as well, but I ran into quite a few issues pretty quickly (https://github.com/bigskysoftware/htmx/issues/689, https://github.com/bigskysoftware/htmx/issues/1015, to name a few). A friend of mine built a few websites using Unpoly, but has since also switched to using swup.

I personally have had a great experience with combining swup + [Alpine.js](https://alpinejs.dev/) + [Alpine Ajax](https://alpine-ajax.js.org/). These three combined give me all the power I need for my projects and I haven't been missing anything since. I also like the focus on progressive enhancement and [accessibility](https://swup.js.org/plugins/a11y-plugin/) in swup. That's  something you would need to be very careful not to break when using htmx or Unpoly.

**Reply by aolko** [reply]
So it would be nice if swup would have a `noConflict` or something similar, since I find htmx pretty handy.

**Reply by daun** [reply]
@aolko In that case, it's probably best if you stick with just htmx. They do similar things, and if you're happy with one, there's little sense in complicating things by combining them. Not sure how a no-conflict option would work, considering the vast amount of available frontend libraries. Feel free to report back if you have specific ideas around what's causing conflicts.
